stable/fluentd/templates/service.yaml

```diff
@@ -15,10 +15,10 @@ spec:
       port: {{ $port.containerPort }}
       targetPort: {{ $port.containerPort }}
       protocol: {{ $port.protocol }}
+      {{- if (and (eq $.Values.service.type "NodePort") (not (empty $port.nodePort))) }}
+      nodePort: {{ $port.nodePort }}
+      {{- end }}
   {{- end }}
-  {{ if (and (eq .Values.service.type "NodePort") (not (empty .Values.service.nodePort))) }}
-      nodePort: {{ .Values.service.nodePort }}
```

Comment:
So what happens to people using  `service.nodePort`?

Reply:
Please check out #12413, I have documented the problem with nodePort.

Reply:
OK, but if I use `--set service.nodePort=...` then this is not backwards compatible since we are dropping the `service.nodePort` flag?

I'm fine with that, but I want to make sure we make users aware of it if it is the case.

Reply:
Yes you are right this will break compatibility with older charts.

Reply:
Therefore, it's either a `2.0.0` or we need a way to keep the old thing.